### ISSUE TITLE
[A11yAudit] Date Picker to be Accessible [III]

### ISSUE THREAD

**Issue by ssabo1** [sparkdesignsystem/Backlog#1322]
**User Story**
As Spark we want Sparks Date Picker to be accessible.

**Notes:**
Paciello Audit on Spark Design System
QLC-11 INPUT - Date Picker
Inaccessible date picker

ISSUE DESCRIPTION
The date picker widget in samples present a number of accessibility barriers that are likely to impact on the ability of certain user groups to complete related tasks.

Presence of date picker widget unannounced
Actions are not keyboard accessible
Date picker widget lacks semantic structure
Controls lack meaningful labels
Zooming/resizing text results in loss of functionality
Issues with content appearing on focus
Form controls rely on placeholder text as a label
Color contrast issues

**RECOMMENDATIONS**
Unfortunately, there is no standardized design pattern for how a date picker should be structured or what types of (keyboard) interactions should be supported. This is particularly compounded by the fact that, while many browsers now support the HTML5 input type="date", which causes the browser's native date picker widget to appear when the field receives focus, these date pickers have a different visual appearance across different browsers, cannot be easily styled, and (depending on the browser) are often just as inaccessible as custom date picker widgets.

We strongly recommend that you consider replacing the current implementation with a date picker that has been designed with accessibility in mind. The External resources section below provides links to some examples.

If you are unable to replace the current implementation, we recommend that you follow the guidance provided in the relevant technique indicated in the Techniques section below to ensure that it can be operated using the keyboard alone, and that it is exposed correctly to assistive technologies.
Make sure that the date picker can be operated using the keyboard alone

We recommend that you provide a keyboard accessible trigger control next to the text input, that expands the date picker widget when activated. This approach will avoid the date picker appearing/disappearing when the text input gains/loses focus. An aria-expanded attribute conveys the state (collapsed/expanded) of the date picker widget. It is assumed that an icon would be used as the trigger.

**AC**

**Comment by ssabo1** [comment]
How do we want to move forward with datepicker?

**Comment by ssabo1** [comment]
Confirmed: This has been added to Spark Design System Team Backlog.
